### PR TITLE
SQL: Fix txt format for empty result sets

### LINKED ISSUE #83371: SQL: Text formatter fails on empty result pages
### Elasticsearch Version

8.1.0-SNAPSHOT (but also affects 7.x releases)

### Installed Plugins

_No response_

### Java Version

_bundled_

### OS Version

Os X

### Problem Description

SQL scroll requests with `format=txt` fail when fetching a subsequent empty page with

```
HTTP/1.1 500 Internal Server Error
X-elastic-product: Elasticsearch
content-type: application/json; charset=UTF-8
content-encoding: gzip
content-length: 148

{
  "error": {
    "root_cause": [
      {
        "type": "sql_illegal_argument_exception",
        "reason": "Cannot find text formatter - this is likely a bug"
      }
    ],
    "type": "sql_illegal_argument_exception",
    "reason": "Cannot find text formatter - this is likely a bug"
  },
  "status": 500
}
```

Note, this only happens for scroll requests. If the first page is already empty (filter matches no records) an empty table is returned as expected.

### Steps to Reproduce

Given the `test_emp` dataset, issue the following requests:

```
POST http://localhost:9201/_sql?format=txt
Content-Type: application/json

{
    "fetch_size": 90,
    "query": "select * from test_emp order by 1"
}
```

Returns the first 90 records and cursor A.

```
POST http://localhost:9201/_sql?format=txt
Content-Type: application/json

{
    "cursor": <insert cursor A from previous response>
}
```

Returns the remaining 10 records and cursor B (see also https://github.com/elastic/elasticsearch/issues/75528).

```
POST http://localhost:9201/_sql?format=txt
Content-Type: application/json

{
    "cursor": <insert cursor B from previous response>
}
```

Fails with the `sql_illegal_argument_exception` mentioned above.

### Logs (if relevant)

_No response_

### PR DESCRIPTION
**Pull request by Luegg**
Addresses #83371

The issue is caused by `org/elasticsearch/xpack/sql/plugin/TextFormatterCursor.java:57` which unwraps empty cursors in `TextFormatterCursor` before being processed by `TextFormat.PLAIN_TEXT.format()`.

### CODE REVIEW

**Review by costin — APPROVED** [review]
LGTM. Since it's an easy bug fix that manifest nasty (when it does) please backport it to 7.x as well.

File: x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/plugin/TextFormatTests.java [hunk 1]
```diff
@@ -163,9 +165,26 @@ public void testInvalidCsvDelims() {
...
+    public void testPlainTextEmptyCursorWithColumns() {
+        assertEquals("""
+                 name     \s
+            ---------------
+            """, PLAIN_TEXT.format(req(), emptyData()));
+    }
+
+    public void testPlainTextEmptyCursorWithoutColumns() {
+        assertEquals(
+            "".stripTrailing(),
```
Comment (bpintea) [inline x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/plugin/TextFormatTests.java:177]: StringUtils.EMPTY?

**Review by bpintea — APPROVED** [review]
LGTM

**Comment by Luegg** [comment]
> LGTM. Since it's an easy bug fix that manifest nasty (when it does) please backport it to 7.x as well.

Good point 👍

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/TextFormat.java [hunk 2]
```diff
@@ -79,6 +79,9 @@ String format(RestRequest request, SqlQueryResponse response) {
             } else if (response.hasId()) {
                 // an async request has no results yet
                 return StringUtils.EMPTY;
+            } else if (response.rows().isEmpty()) {
+                // no data and no headers to return
+                return StringUtils.EMPTY;
             }
```
Comment (astefan) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/TextFormat.java:85]:
```suggestion
            } else if (response.hasId() || response.rows().isEmpty()) {
                // for an async request that has no results yet or for an emtpy page, there is nothing to return (no data and no headers)
                return StringUtils.EMPTY;
            }
```

Just integrate this `else if` branch in the previous one, keeping the comment.

**Review by astefan — APPROVED** [review]
LGTM
Added one small suggestion, but it's also  ok with me to keep your original approach.

**Pull request merged**